pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -2358,166 +2341,229 @@ func validateDisks(field *k8sfield.Path, disks []v1.Disk) []metav1.StatusCause {
 
 		// Verify bus is supported, if provided
 		if len(bus) > 0 {
-			if bus == "ide" {
-				causes = append(causes, metav1.StatusCause{
-					Type:    metav1.CauseTypeFieldValueInvalid,
-					Message: "IDE bus is not supported",
-					Field:   field.Index(idx).Child(diskType, "bus").String(),
-				})
-			} else {
-				buses := []v1.DiskBus{v1.DiskBusVirtio, v1.DiskBusSCSI, v1.DiskBusSATA, v1.DiskBusUSB}
-				validBus := false
-				for _, b := range buses {
-					if b == bus {
-						validBus = true
-					}
-				}
-				if !validBus {
-					causes = append(causes, metav1.StatusCause{
-						Type:    metav1.CauseTypeFieldValueInvalid,
-						Message: fmt.Sprintf("%s is set with an unrecognized bus %s, must be one of: %v", field.Index(idx).String(), bus, buses),
-						Field:   field.Index(idx).Child(diskType, "bus").String(),
-					})
-				}
+			causes = append(causes, validateBusSupported(disk, bus, diskType, field, idx)...)
```

Comment:
I would prefer that the `len(bus)` control be in the `validateBusSupported` function with an early exit.
i.e.
```
func validateBusSupported(disk v1.Disk, bus v1.DiskBus, diskType string, field *k8sfield.Path, idx int) (causes []metav1.StatusCause) {
	if len(bus) == 0 {
		return
	}
	
	if bus == "ide" {
```